pkgs/development/libraries/eclib/default.nix

```diff
@@ -14,7 +14,7 @@ assert withFlint -> flint != null;
 
 stdenv.mkDerivation rec {
   pname = "eclib";
-  version = "20210223"; # upgrade might break the sage interface
+  version = "20190909"; # upgrade might break the sage interface
```

Comment:
I think a comment that explains the reason for the outdated version would have been useful. It could prevent "accidental" updates in the future.

@collares since you [have](https://github.com/NixOS/nixpkgs/commit/ef615b5a61c049317da5ed5637e4561b224dd66d) [done](https://github.com/NixOS/nixpkgs/pull/116365#issuecomment-821884448) most of the work for the update already it might be a good idea to create an upstream ticket. That way others could benefit from the update and we wouldn't have to pin the version. The sage contribution culture is pretty collaborative, so they would probably appreciate a ticket even if you don't include the "sage-the-distribution" specific changes. Those are a bit difficult to test on nixos. Somebody else might pick it up. What do you think?

Reply:
Way ahead of you, see https://github.com/JohnCremona/eclib/issues/64 :)

When updating Sage 9.2 to eclib 20210226, I discovered that 20210223 was not an "actual" release (see above). Upstream just bumps versions sometimes, but that does not mean it is a new release unless it appears at https://github.com/JohnCremona/eclib/releases. Although 20210223 is not a release, there are newer releases of eclib, so perhaps I should have updated to 20210415 (say). However, the required Sage patches (tracked at https://trac.sagemath.org/ticket/31443) introduce API changes, so I am basically waiting for them to land before updating eclib. This will happen after 9.3.

Reply:
Ah I see, so you were already working with upstream. In that case only the comment regarding a comment ( ;) ) remains, but that will probably [soon](https://github.com/nixos/nixpkgs/pull/122592#issuecomment-838648807) be obsolete anyway. If you do end up opening a PR for the update, I think it would  be a good idea to add a disclaimer that explains the "version tag vs release" situation.

Good work :)